### ISSUE TITLE
Dashboard filter to nested question

### ISSUE THREAD

**Issue by Crusader4Christ** [metabase/metabase#9802]
To workaround for column ordering was created SQL question, around it was created the pie chart with grouping. Then this pie chart was added to dashboard and connected to `deposited` date filter.
Pie chart do not work on the dashboard but works on the new page (click on the title in dashboard) even `deposited` filter is filled right.

Metabase v0.32.4
Database: MySQL
Here is the response:
```
{"database_id":-1337,"started_at":"2019-04-17T14:44:49.229Z","json_query":{"query":{"source-table":"card__141","aggregation":[["count"]],"breakout":[["field-literal","country_id","type/Text"]],"filter":["=",["field-literal","status","type/Integer"],5]},"type":"query","database":-1337,"constraints":{"max-results":10000,"max-results-bare-rows":2000},"parameters":[{"type":"date/all-options","target":["dimension",["field-literal","deposited","type/DateTime"]],"value":"thisday"}],"middleware":null,"cache-ttl":null},"native":null,"status":"failed","stacktrace":["--> models.params$fn__33153$field_form__GT_id__33158.invoke(params.clj:21)","query_processor.middleware.parameters.mbql$fn__33519$build_filter_clause__33524$fn__33534.invoke(mbql.clj:50)","query_processor.middleware.parameters.mbql$fn__33519$build_filter_clause__33524.invoke(mbql.clj:39)","query_processor.middleware.parameters.mbql$expand.invokeStatic(mbql.clj:72)","query_processor.middleware.parameters.mbql$expand.invoke(mbql.clj:59)","query_processor.middleware.parameters$expand_parameters_STAR_.invokeStatic(parameters.clj:20)","query_processor.middleware.parameters$expand_parameters_STAR_.invoke(parameters.clj:14)","query_processor.middleware.parameters$expand_parameters.invokeStatic(parameters.clj:43)","query_processor.middleware.parameters$expand_parameters.invoke(parameters.clj:40)","query_processor.middleware.parameters$substitute_parameters_STAR_.invokeStatic(parameters.clj:49)","query_processor.middleware.parameters$substitute_parameters_STAR_.invoke(parameters.clj:46)","query_processor.middleware.driver_specific$process_query_in_context$fn__32792.invoke(driver_specific.clj:12)","query_processor.middleware.resolve_driver$resolve_driver$fn__35968.invoke(resolve_driver.clj:15)","query_processor.middleware.bind_effective_timezone$bind_effective_timezone$fn__31988$fn__31989.invoke(bind_effective_timezone.clj:9)","util.date$call_with_effective_timezone.invokeStatic(date.clj:88)","util.date$call_with_effective_timezone.invoke(date.clj:77)","query_processor.middleware.bind_effective_timezone$bind_effective_timezone$fn__31988.invoke(bind_effective_timezone.clj:8)","query_processor.middleware.store$initialize_store$fn__39027$fn__39028.invoke(store.clj:11)","query_processor.store$do_with_new_store.invokeStatic(store.clj:35)","query_processor.store$do_with_new_store.invoke(store.clj:31)","query_processor.middleware.store$initialize_store$fn__39027.invoke(store.clj:10)","query_processor.middleware.cache$maybe_return_cached_results$fn__32410.invoke(cache.clj:127)","query_processor.middleware.catch_exceptions$catch_exceptions$fn__32458.invoke(catch_exceptions.clj:63)","query_processor$process_query.invokeStatic(query_processor.clj:232)","query_processor$process_query.invoke(query_processor.clj:228)","query_processor$run_and_save_query_BANG_.invokeStatic(query_processor.clj:344)","query_processor$run_and_save_query_BANG_.invoke(query_processor.clj:338)","query_processor$fn__39216$process_query_and_save_execution_BANG___39221$fn__39222.invoke(query_processor.clj:371)","query_processor$fn__39216$process_query_and_save_execution_BANG___39221.invoke(query_processor.clj:357)","async.semaphore_channel$do_f_with_permit$f__46608.invoke(semaphore_channel.clj:81)","async.util$do_on_separate_thread$fn__42559.invoke(util.clj:53)"],"context":"question","error":"(not (integer? \"deposited\"))","row_count":0,"running_time":676,"data":{"rows":[],"cols":[],"columns":[]}}
```
:arrow_down: Please click the :+1: reaction instead of leaving a `+1` or `update?` comment

**Comment by flamber** [comment]
Confirmed on 0.32.4 with H2 Sample Dataset with Nested Question from Custom Question.

**Comment by chucksense** [comment]
Steps to reproduce, per discussion: https://discourse.metabase.com/t/help-with-dashboard-filters/6844/4

Hi flamber,
To reproduce, you can use the sample data set.

First, save a question using this query:

```sql
WITH tmp_user_order_dates as (
  SELECT 
    o.USER_ID,
    o.CREATED_AT,
    o.QUANTITY
  FROM 
    ORDERS o
),

tmp_prior_orders_by_date as (
  select
      tbod.USER_ID,
      tbod.CREATED_AT,
      tbod.QUANTITY,
      (select count(*) from tmp_user_order_dates tbod2 where tbod2.USER_ID = tbod.USER_ID and tbod2.CREATED_AT < tbod.CREATED_AT ) as PRIOR_ORDERS
  from tmp_user_order_dates tbod
)

select
  date_trunc('day', tpobd.CREATED_AT) as "Date",
  case when tpobd.PRIOR_ORDERS > 0 then 'Return' else 'New' end as "Customer Type",
  sum(QUANTITY) as "Items Sold"
from tmp_prior_orders_by_date tpobd
group by date_trunc('day', tpobd.CREATED_AT), "Customer Type"
order by date_trunc('day', tpobd.CREATED_AT) asc
```

Then configure the second question referencing the first, like this:

![](https://discourse.metabase.com/uploads/default/original/2X/0/0536bfd451962933af0e230dac9d8220a5024281.jpeg)


Then when you add it to a dashboard and configure a date filter, once a filter is applied, you can see the error:

![](https://discourse.metabase.com/uploads/default/original/2X/b/bb1e992564ad8822cbb053c411867674e044a6e6.jpeg)

**Comment by Juanses** [comment]
@chucksense Did you find a solution for your problem ?
I'm having the same, I can't filter a nested question

**Comment by chucksense** [comment]
@Juanses No, sorry, still not working for me and I gave up trying to work around it.

**Comment by tlrobinson** [comment]
@camsaul I think this is a backend thing. It seems like we don't support parameters on nested queries / `field-literal` dimensions.

I followed the repro instructions above, and the dashboard has one parameter like:

```js
{
  name: "Relative Date",
  slug: "relative_date",
  id: "e4ee6f31",
  type: "date/relative"
}
```

and the dashcard has a parameter mapping like:

```js
{
  card_id: 184,
  parameter_id: "e4ee6f31",
  target: ["dimension", ["field-literal", "Date", "type/DateTime"]]
}
```

then when the query API call for that card is made it has the following payload:

```js
{
  parameters: [
    {
      type: "date/relative",
      target: ["dimension", ["field-literal", "Date", "type/DateTime"]],
      value: "thisyear"
    }
  ]
}
```

all of which seems correct to me.

The resulting stack trace looks like this:

```
09-09 15:20:17 WARN middleware.process-userland-query :: Query failure {:status :failed,
 :class clojure.lang.ExceptionInfo,
 :error (not (integer? "Date")),
 :stacktrace
 ("--> models.params$eval32218$field_form__GT_id__32223.invoke(params.clj:21)"
  "query_processor.middleware.parameters.mbql$eval44715$build_filter_clause__44720$fn__44730.invoke(mbql.clj:50)"
  "query_processor.middleware.parameters.mbql$eval44715$build_filter_clause__44720.invoke(mbql.clj:39)"
  "query_processor.middleware.parameters.mbql$expand.invokeStatic(mbql.clj:78)"
  "query_processor.middleware.parameters.mbql$expand.invoke(mbql.clj:65)"
  "query_processor.middleware.parameters$expand_mbql_params.invokeStatic(parameters.clj:39)"
  "query_processor.middleware.parameters$expand_mbql_params.invoke(parameters.clj:35)"
  "query_processor.middleware.parameters$expand_one.invokeStatic(parameters.clj:63)"
  "query_processor.middleware.parameters$expand_one.invoke(parameters.clj:56)"
  "query_processor.middleware.parameters$expand_all$replace_47109__47110.invoke(parameters.clj:74)"
  "mbql.util.match$replace_in_collection$iter__25857__25861$fn__25862.invoke(match.clj:132)"
  "mbql.util.match$replace_in_collection.invokeStatic(match.clj:131)"
  "mbql.util.match$replace_in_collection.invoke(match.clj:126)"
  "query_processor.middleware.parameters$expand_all$replace_47109__47110.invoke(parameters.clj:74)"
  "query_processor.middleware.parameters$expand_all.invokeStatic(parameters.clj:74)"
  "query_processor.middleware.parameters$expand_all.invoke(parameters.clj:68)"
  "query_processor.middleware.parameters$expand_all.invokeStatic(parameters.clj:71)"
  "query_processor.middleware.parameters$expand_all.invoke(parameters.clj:68)"
  "query_processor.middleware.parameters$expand_parameters.invokeStatic(parameters.clj:92)"
  "query_processor.middleware.parameters$expand_parameters.invoke(parameters.clj:88)"
  "query_processor.middleware.parameters$substitute_parameters_STAR_.invokeStatic(parameters.clj:97)"
  "query_processor.middleware.parameters$substitute_parameters_STAR_.invoke(parameters.clj:94)"
  "query_processor.middleware.driver_specific$process_query_in_context$fn__43889.invoke(driver_specific.clj:12)"
  "query_processor.middleware.resolve_driver$resolve_driver$fn__47488.invoke(resolve_driver.clj:22)"
  "query_processor.middleware.bind_effective_timezone$bind_effective_timezone$fn__43000$fn__43001.invoke(bind_effective_timezone.clj:9)"
  "util.date$call_with_effective_timezone.invokeStatic(date.clj:88)"
  "util.date$call_with_effective_timezone.invoke(date.clj:77)"
  "query_processor.middleware.bind_effective_timezone$bind_effective_timezone$fn__43000.invoke(bind_effective_timezone.clj:8)"
  "query_processor.middleware.store$initialize_store$fn__50992$fn__50993.invoke(store.clj:11)"
  "query_processor.store$do_with_store.invokeStatic(store.clj:46)"
  "query_processor.store$do_with_store.invoke(store.clj:40)"
  "query_processor.middleware.store$initialize_store$fn__50992.invoke(store.clj:10)"
  "query_processor.middleware.async$async__GT_sync$fn__40204.invoke(async.clj:23)"
  "query_processor.middleware.async_wait$runnable$fn__42714.invoke(async_wait.clj:89)"),
 :query
 {:constraints {:max-results 10000, :max-results-bare-rows 2000},
  :type :query,
  :middleware {:userland-query? true},
  :info
  {:executed-by 1,
   :context :question,
   :card-id 184,
   :dashboard-id nil,
   :query-hash [-106, -24, -95, -104, -77, -113, -3, 41, 30, 78, 67, 89, -45, -90, 93, 37, 62, -18, 97, 92, 40, -99, -125, 98, 51, -24, -30, 51, -51, 17, -78, 92]},
  :query
  {:source-table "card__183",
   :aggregation [[:sum [:field-literal "Items Sold" :type/Integer]]],
   :breakout [[:datetime-field [:field-literal "Date" :type/DateTime] :week] [:field-literal "Customer Type" :type/Text]]},
  :parameters [{:type "date/relative", :target ["dimension" ["field-literal" "Date" "type/DateTime"]], :value "thisyear"}],
  :async? true,
  :cache-ttl 540},
 :preprocessed nil,
 :native nil,
 :ex-data {:type :schema.core/error, :value "Date", :error (not (integer? "Date"))}}
```

This TODO in `src/metabase/models/params.clj` suggests `field-literal` support hasn't been completed:

```clojure
(s/defn field-form->id :- su/IntGreaterThanZero
  "Expand a `field-id` or `fk->` `form` and return the ID of the Field it references. Also handles unwrapped integers.

    (field-form->id [:field-id 100]) ; -> 100"
  [field-form]
  (if (integer? field-form)
    field-form
    ;; TODO - what are we supposed to do if `field-form` is a field literal?
    (mbql.u/field-clause->id-or-literal field-form)))
```

**Comment by mazameli** [comment]
Based on @tlrobinson's comment, it sounds like this is actually a missing/incomplete feature rather than a bug. `field-literal` support hasn't been implemented for parameters yet. Relabeling this as a `Completeness` issue.

**Comment by flamber** [comment]
This is probably a duplicate of #6310, but this has more activity and upvotes.

**Comment by toddejohnson** [comment]
Am I running into something different?

On mine it works if you refresh(F5) the dashboard but not when you view(click link) dashboard. When you click on the question on the dashboard it works just fine.

**Comment by flamber** [comment]
@toddejohnson You're most likely seeing something different. I would recommend that you head over to the forum ( https://discourse.metabase.com/ ) to troubleshoot - please include Diagnostic Info from Admin > Troubleshooting and steps to reproduce the problem.

**Comment by ferarnoled** [comment]
Any updates on this? Didn't see this ticket on the roadmap

**Comment by mazameli** [comment]
@ferarnoled Yes, it's just recently been prioritized.

**Comment by camsaul** [comment]
Implemented by #12531
